tests/slow/src/test/scala/tests/sbt/SbtServerSuite.scala

```diff
@@ -240,4 +247,30 @@ class SbtServerSuite
       )
     }
   }
+
+  test("infinite-loop") {
+    cleanWorkspace()
+    val layout =
+      s"""|/project/build.properties
+          |sbt.version=$supportedMetaBuildVersion
+          |/build.sbt
+          |${SbtBuildLayout.commonSbtSettings}
+          |Compile / sourceGenerators += Def.task {
+          |  val content = "object Foo\\n"
+          |  val file = target.value / "Foo.scala"
+          |  IO.write(file, content)
+          |  Seq(file)
+          |}
+          |""".stripMargin
+    for {
+      _ <- initialize(layout)
+    } yield {
+      // Sleep 5s: that should be enough to see the compilation looping
+      // more than once
+      Thread.sleep(5000)
```

Comment:
It would be awesome if we could work around having to wait 5 seconds, but not sure how to do it currently. We could potenatially run `didSave` on an additional file manually and see how many files are recompiled.

But if there is no better way that's fine as it is.

Reply:
If I force the first compilation, then I think I can safely reduce the sleep to 100ms.